#!/usr/bin/env python
PACKAGE = "jsk_robot_startup"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("mean_x", double_t, 0, "Average for linear x", 0.0, -10.0, 10.0)
gen.add("mean_y", double_t, 0, "Average for linear y", 0.0, -10.0, 10.0)
gen.add("mean_z", double_t, 0, "Average for linear z", 0.0, -10.0, 10.0)
gen.add("mean_roll", double_t, 0, "Average for angular x", 0.0, -10.0, 10.0)
gen.add("mean_pitch", double_t, 0, "Average for angular y",  0.0, -10.0, 10.0)
gen.add("mean_yaw", double_t, 0, "Average for angular z", 0.0, -10.0, 10.0)
gen.add("sigma_x", double_t, 0, "Standard diviation for linear x", 0.05, 1e-6, 10.0)
gen.add("sigma_y", double_t, 0, "Standard diviation for linear y", 0.05, 1e-6, 10.0)
gen.add("sigma_z", double_t, 0, "Standard diviation for linear z", 0.05, 1e-6, 10.0)
gen.add("sigma_roll", double_t, 0, "Standard diviation for angular x", 0.05, 1e-6, 10.0)
gen.add("sigma_pitch", double_t, 0, "Standard diviation for angular y", 0.05, 1e-6, 10.0)
gen.add("sigma_yaw", double_t, 0, "Standard diviation for angular z", 0.05, 1e-6, 10.0)

exit(gen.generate(PACKAGE, "constant_height_frame_publisher_reconfigure", "OdometryOffsetReconfigure"))
